### PR TITLE
first pass

### PR DESCRIPTION
**Pull request by ardz**

### CODE REVIEW

File: src/Aveva.Weather.WebApi/Service/WeatherService.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿using Aveva.Weather.Models.Domain;
+using Newtonsoft.Json;
+
+namespace Aveva.Weather.WebApi.Service;
+
+public class WeatherService : IWeatherService
+{
+    public  WeatherForecast? UpdateForecast()
```
Comment (fauxcoding) [inline src/Aveva.Weather.WebApi/Service/WeatherService.cs:8]: Consider changing the name as it suggests a state change is being made. I.e. `GetForecast`

File: src/Aveva.Weather.WebApi/Service/WeatherService.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        var weatherJson = string.Empty;
+
+        try
+        {
+            weatherJson = File.ReadAllText(filePath);
+        }
+        catch (FileNotFoundException)
+        {
+            Console.WriteLine("The JSON file does not exist.");
```
Comment (fauxcoding) [inline src/Aveva.Weather.WebApi/Service/WeatherService.cs:20]: Consider removing these statements as your project is a webapp not a console app.

File: test/Aveva.Weather.WebApi.Tests/WeatherTests.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        {
+            Assert.Equal(0, precipitationValues2.Precipitation.Inches);
+            Assert.Equal(0, precipitationValues2.Precipitation.Millimeters);
+        }
+    }
+
+    // "There are the correct number of entries for London"
+    // this requirement is ambiguous
+    // in the real world, wouldn't work on this until it was clarified what it means 
+    [Fact]
```
Comment (fauxcoding) [inline test/Aveva.Weather.WebApi.Tests/WeatherTests.cs:85]: I think it means you need to check `"day_range":5,` and then assert that there are 5 forecasts.

Comment (ardz) [reply test/Aveva.Weather.WebApi.Tests/WeatherTests.cs:85]: hadn't thought of that, cheers man much appreciated.

**Comment by ardz** [comment]
DONE

**Pull request merged**